include/ignition/gazebo/comms/MsgManager.hh

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_GAZEBO_MSGMANAGER_HH_
+#define IGNITION_GAZEBO_MSGMANAGER_HH_
+
+#include <deque>
+#include <map>
+#include <memory>
+#include <string>
+
+#include <ignition/transport/Node.hh>
+#include <ignition/utils/ImplPtr.hh>
+#include "ignition/gazebo/config.hh"
+#include "ignition/gazebo/System.hh"
+
+namespace ignition
+{
+namespace msgs
+{
+  // Forward declarations.
+  class Dataframe;
+}
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace comms
+{
+
+/// \brief A queue of message pointers.
+using DataQueue = std::deque<msgs::DataframeSharedPtr>;
+
+/// \brief A map where the key is the topic subscribed to an address and
+/// the value is a publisher to reach that topic.
+using SubscriptionHandler = std::map<std::string, transport::Node::Publisher>;
```

Comment:
> > I've done a cursory review of this. Overall its looking pretty good. I spotted some places where we used `std::map` instead of `std::unordered_map`. Since this is targeting fortress and I believe we are using C++17 we should prefer `unordered_map` unless we need the data sorted. One issue I have is with the Registry type. Currently its an alias of an `std::map` however as I've listed this leads to two problems (1) fragile API/ABI since this is something end users implementing comms plugins will use making it difficult for us to change underlying implementation and (2) the user has to access and internal data type to figure out how to add a message to the outbound. I can make a PR targeting this PR with the relevant changes.
> > I think as we discussed before the comms model for acoustics will need the ability to be stepped at a different pace. That can be addressed in a future PR.
> 
> Thanks for the suggestions and the offer to help! They all sound pretty good. If you're doing the Registry transformation from an alias to its own class I can work in parallel in the changes to step at different frequency than physics.

The ability to specify `<step_size>` has been added in [beefb6b](https://github.com/ignitionrobotics/ign-gazebo/pull/1416/commits/beefb6bf950dc48ffb783c2c17ca2357ca8ad846) (mostly copied from LRUAV :smile: )